src/tests/ftest/io/llnl_mpi4py.py

```diff
@@ -0,0 +1,176 @@
+import AgentUtils
+import server_utils
+import write_host_file
+from mpio_utils import MpioUtils, MpioFailed
+from daos_api import DaosContext, DaosPool, DaosApiError
+
+class LlnlMpi4py(Test):
+    """
+    Runs LLNL and MPI4PY test suites.
+    """
+
+    def setUp(self):
+        # initialising variables
+        self.basepath = None
+        self.server_group = None
+        self.context = None
+        self.pool = None
+        self.mpio = None
+        self.agent_sessions = None
+        self.hostlist_servers = None
+        self.hostfile_servers = None
+        self.hostlist_clients = None
+        self.hostfile_clients = None
+        self.createsvc = None
+
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as var_file:
+            build_paths = json.load(var_file)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+
+        self.server_group = self.params.get("name", '/server_config/',
+                                            'daos_server')
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/')
+        self.hostfile_servers = write_host_file.write_host_file(
+            self.hostlist_servers, self.workdir)
+        print("Host file servers is: {}".format(self.hostfile_servers))
+
+        self.hostlist_clients = self.params.get("test_clients", '/run/hosts/')
+        self.hostfile_clients = write_host_file.write_host_file(
+            self.hostlist_clients, self.workdir, None)
+        print("Host file clients is: {}".format(self.hostfile_clients))
+
+        self.agent_sessions = AgentUtils.run_agent(self.basepath,
+                                                   self.hostlist_servers,
+                                                   self.hostlist_clients)
+        # start servers
+        server_utils.run_server(self.hostfile_servers, self.server_group,
+                                self.basepath)
+        try:
+            # parameters used in pool create
+            createmode = self.params.get("mode", '/run/pool/createmode/*/')
+            createuid = os.geteuid()
+            creategid = os.getegid()
+            createsetid = self.params.get("setname", '/run/pool/createset/')
+            createsize = self.params.get("size", '/run/pool/createsize/')
+            self.createsvc = self.params.get("svcn", '/run/pool/createsvc/')
+
+            # initialize a python pool object then create the underlying
+            # daos storage
+            self.pool = DaosPool(self.context)
+            self.pool.create(createmode, createuid, creategid,
+                             createsize, createsetid, None, None,
+                             self.createsvc)
+        except (DaosApiError) as excep:
+            self.fail("<Test Failed at pool create> \n{1}".format(excep))
+
+    def tearDown(self):
```

Comment:
Okay, I'm sorry for adding another comment.  This could be fixed later.  I don't think this teardown is bullet proof